packages/shared/lib/core/wallet/utils/outputs/isOutputExpired.ts

```diff
@@ -0,0 +1,21 @@
+import { CommonOutput, ExpirationUnlockCondition, UnlockConditionType } from '@iota/sdk/out/types'
+import { get } from 'svelte/store'
+import { getSlotIndexFromNodeInfo, nodeInfoProtocolParameters } from 'shared/lib/core/network'
+
+export function isOutputExpired(output: CommonOutput): boolean | undefined {
+    const expirationUnlockCondition = output.unlockConditions.find(
+        (unlockCondition) => unlockCondition.type === UnlockConditionType.Expiration
+    ) as ExpirationUnlockCondition
+    const nodeProtocolParameters = get(nodeInfoProtocolParameters)
+    const outputSlotIndex = expirationUnlockCondition?.slotIndex
+    if (!nodeProtocolParameters || !outputSlotIndex) return true
+
+    const actualSlotIndex = getSlotIndexFromNodeInfo(nodeProtocolParameters)
+    if (actualSlotIndex > outputSlotIndex + nodeProtocolParameters.maxCommittableAge) {
+        return false
+    } else if (actualSlotIndex <= outputSlotIndex + nodeProtocolParameters.minCommittableAge) {
+        return true
+    } else {
+        return
+    }
```

Comment:
Just check if it is expired, if it's not, return a false, I don't see why returning `undefined` is needed

```suggestion
    return actualSlotIndex <= outputSlotIndex + nodeProtocolParameters.minCommittableAge
```